rpc/grpc/client/block_service.go

```diff
@@ -0,0 +1,100 @@
+package client
+
+import (
+	"context"
+
+	blocksvc "github.com/cometbft/cometbft/proto/tendermint/services/block/v1"
+	"github.com/cometbft/cometbft/types"
+	"github.com/cosmos/gogoproto/grpc"
+)
+
+// Block data returned by the CometBFT BlockService gRPC API.
+type Block struct {
+	BlockID types.BlockID `json:"block_id"`
+	Block   *types.Block  `json:"block"`
+}
+
+// BlockServiceClient provides block information
+type BlockServiceClient interface {
+	GetBlockByHeight(ctx context.Context, height int64) (*Block, error)
+	// GetLatestHeight provides sends the latest committed block height to the given output
+	// channel as blocks are committed.
+	GetLatestHeight(ctx context.Context, out chan<- int64) error
+}
+
+type blockServiceClient struct {
+	client blocksvc.BlockServiceClient
+}
+
+func newBlockServiceClient(conn grpc.ClientConn) BlockServiceClient {
+	return &blockServiceClient{
+		client: blocksvc.NewBlockServiceClient(conn),
+	}
+}
+
+// GetBlockByHeight implements BlockServiceClient GetBlockByHeight
+func (c *blockServiceClient) GetBlockByHeight(ctx context.Context, height int64) (*Block, error) {
+	req := blocksvc.GetByHeightRequest{
+		Height: height,
+	}
+	res, err := c.client.GetByHeight(ctx, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert Block from proto to core type
+	block, err := types.BlockFromProto(res.Block)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert BlockID from proto to core type
+	blockID, err := types.BlockIDFromProto(res.BlockId)
+	if err != nil {
+		return nil, err
+	}
+
+	response := Block{
+		BlockID: *blockID,
+		Block:   block,
+	}
+	return &response, nil
+}
+
+// GetLatestHeight implements BlockServiceClient GetLatestHeight
+func (c *blockServiceClient) GetLatestHeight(ctx context.Context, ch chan<- int64) error {
+	req := blocksvc.GetLatestHeightRequest{}
+
+	latestHeight, err := c.client.GetLatestHeight(ctx, &req)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			response, err := latestHeight.Recv()
+			if err != nil {
+				break
+			}
+			ch <- response.Height
```

Comment:
```suggestion
			select {
			case ch <- response.Height:
			default:
			}
```

As per my previous question, what's a good behaviour to implement if the `out` channel's full? We should check for this and probably just drop the publishing of the current height so as to not block this goroutine. They'll get the next height anyways once the channel has space again.

If we go this route we should probably document the behaviour in the docstring so callers are aware that they may get non-contiguous heights.